### PR TITLE
Add overlap operator

### PR DESCRIPTION
**Pull request by cliffburdick**
Added `overlap` operator for counterpart to tensor `OverlapView`

### CODE REVIEW

File: docs_input/api/manipulation/rearranging/overlap.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+.. _overlap_func:
+
+overlap
+#######
+
+Create an overlapping view an of input operator giving a higher-rank view of the input
+
+For example, the following 1D tensor [1 2 3 4 5] could be cloned into a 2d tensor with a
+window size of 2 and overlap of 1, resulting in:
+ [1 2
```
Comment (tmartin-gh) [inline docs_input/api/manipulation/rearranging/overlap.rst:10]: Doesn't render the rows correctly.

File: test/00_operators/OperatorTests.cu [hunk 2]
```diff
@@ -2378,6 +2378,84 @@ TYPED_TEST(OperatorTestsNumericAllExecs, Reshape)
...
+  using TestType = std::tuple_element_t<0, TypeParam>;
+  using ExecType = std::tuple_element_t<1, TypeParam>;
+  using inner_type = typename inner_op_type_t<TestType>::type;
+
+  ExecType exec{}; 
+
+
+  tensor_t<TestType, 1> a{{10}};
+  a.SetVals({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+  auto ao = overlap(a, {4}, {2});
```
Comment (tmartin-gh) [inline test/00_operators/OperatorTests.cu:2394]: Perhaps also have an ASSERT_EQ on the Shape of ao?  If there is a code error and ao had a larger shape than tested by the double-for loop below, it wouldn't be caught.

File: docs_input/api/manipulation/rearranging/overlap.rst (hunk 1 above)
Comment (cliffburdick) [reply docs_input/api/manipulation/rearranging/overlap.rst:10]: This is an rst file, so it should just be newlines as it's shown there

**Review by tbensonatl — APPROVED** [review]

**Pull request merged**